Fix selection not being reset correct when changing between rulesets

peppy opened the pull request:
Carousels filtered to results with no maps visible were not being handled correctly in a few different ways. This covers all those scenarios.

peppy committed:
Fix selection not being reset correct when changing between rulesets

Carousels filtered to results with no maps visible were not being handled correctly in a few different ways. This covers all those scenarios.

Tom94 commented on osu.Game/Screens/Select/BeatmapCarousel.cs:
```diff
@@ -133,7 +133,7 @@ public void SelectBeatmap(BeatmapInfo beatmap, bool animated = true)
         public void SelectNext(int direction = 1, bool skipDifficulties = true)
         {
-            if (groups.Count == 0)
+            if (groups.Count == 0 || groups.All(g => g.State == BeatmapGroupState.Hidden))
```
Can do `!groups.Any(g => g.State != BeatmapGroupState.Hidden)`. Then you save yourself the `Count == 0` check.

peppy replied:
I think I did this, and resharper didn't like it

peppy committed:
Remove unnecessary count check

peppy committed:
Switch to correct ruleset when changing beatmap

This is only really noticeable when using the MusicController to change tracks while at song select.

Tom94 committed:
Merge branch 'master' into fix-empty-carousel

Tom94 commented:
Our good friend, the ImportBeatmapOverIPC test case. \o/

Tom94 committed:
Merge branch 'master' into fix-empty-carousel

Tom94 committed:
Merge branch 'master' into fix-empty-carousel
